### ISSUE TITLE
[FEATURE] Consider ERB indentation

### ISSUE THREAD

**Issue by Splines** [yeonjuan/html-eslint#422]
HTML ESLint supports integrating a template engine like Ruby ERB, which works like a charm. It would be amazing if indentation could also be detected and exposed via a config option. Consider this example:

```erb
<%= form_with(model: @event) do |f| %>
<div class="form-floating">
  <%= f.text_field :name, required: true, class: "form-control", placeholder: "" %>
  <%= f.label :name %>
</div>
<% end %>
```

A config option could allow us to specify that we want indentation here as well:

```erb
<%= form_with(model: @event) do |f| %>
  <div class="form-floating">
    <%= f.text_field :name, required: true, class: "form-control", placeholder: "" %>
    <%= f.label :name %>
  </div>
<% end %>
```

### Possible problems

- I'm not sure if this issue is something html-eslint can provide since you'd probably have to use a whole ERB parser to detect these scenarios holistically. E.g. consider already that not all ERB expressions require nesting, e.g. a simple variable assignment is a one-liner.
- Also, nesting should be properly detected, e.g. imagine another ERB block inside the `form_with` block.

**Comment by yeonjuan** [comment]
Hi @Splines. Thanks for the suggestion. That’s a good idea. I plan to evolve the parser so that it can also determine whether a template is block-level or inline, and whether it is conditional. At that point, I expect it will also be able to handle indentation for blocks.

**Comment by JustNoon** [comment]
This is also the case for Perl's Template Toolkit 2 (just wanted to bring it up as examples for generalization)
```html
<!-- Template Toolkit 2, Perl-like -->
[% FOREACH row IN rows %]
  <tr>
  [% FOREACH td in row %]
    <td>
      [% td %]
    </td>
  [% END %]
[% END %]
```

There is the added challenge that it supports a ton of different syntaxes and styles
```html
<!-- Template Toolkit 2, Perl-like -->
[% SET always_true IF 1 == 1 %] <!-- Blocks don't always end in END -->
[%
  IF always_true;
    output = 'Still true';
  END;
%] <!-- Multiple directives can be delimited with semicolons, including blocks with their own end -->
<span[% always_true ? '><i>*</i>' : '>' %][% output %]</span> <!-- Directives can mangle html structure -->
```

This is also the case for most of the templates of Twine
```html
<!-- Sugarcube 2, javascript-like -->
<<if Date.now() gte $triggerTime || !!window.forceAlert >>
  <span>Time's up</span>
  <<run $lastTrigger = {
    type: 'Date',
    time: Date.now(),
  }>>
<</if>>
```

**Comment by Splines** [comment]
As another project, see [**Herb**](https://herb-tools.dev), which is specifically for HTML & ERB in Ruby/Rails development. It has its own parser, which gets the indentation right, and the integration is really smooth in VSCode. It currently lacks behind in what concerns the number of linting/formatting rules though (which will of course change in the future, so worth to keep an eye on it as well).
